lab-judy/test/gallery-router-test.js

```diff
@@ -0,0 +1,338 @@
+        expect(res.body.userID).to.equal(this.tempUser._id.toString());
+        let date = new Date(res.body.created).toString();
+        expect(date).to.not.equal('Invalid Date');
+        done();
+      });
+    });
+
+    //POST test 401 for unauthorized token
+    it('should return status 401 for no token provided', (done) => {
+      request.post(`${url}/api/gallery`)
+      .send(exampleGallery)
+      .end((err, res) => {
+        expect(res.status).to.equal(401);
+        done();
+      });
+    });
+
+    //POST test 400 for bad request if no body or invalid body provided
+    it('should return 400 for bad request if no body or invalid body provided', (done) => {
+      request.post(`${url}/api/gallery`)
+      .set('Content-type', 'application/json')
+      .set({Authorization: `Bearer ${this.tempToken}`})
+      .send('{')
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        done();
+      });
+    });
+
+
+//GET tests
+    describe('testing GET to /api/gallery/:id', () => {
+      before(done => {
+        console.log('create user');
+        new User(exampleUser)
+      .generatePasswordHash(exampleUser.password)
+      .then( user => user.save())
+      .then( user => {
+        this.tempUser = user;
+        return user.generateToken();
+      })
+      .then( token => {
+        this.tempToken = token;
+        done();
+      })
+      .catch(done);
+      });
+
+      before( done => {
+        console.log('create gallery');
+        exampleGallery.userID = this.tempUser._id.toString();
+        new Gallery(exampleGallery).save()
+      .then( gallery => {
+        this.tempGallery = gallery;
+        done();
+      })
+      .catch(done);
+      });
+
+      after(() => {
+        delete exampleGallery.userID;
+      });
+
+    //GET test 200 to return a gallery
+      it('should return a gallery and status 200', done => {
+        request.get(`${url}/api/gallery/${this.tempGallery._id}`)
+      .set({
+        Authorization: `Bearer ${this.tempToken}`
+      })
+      .end((err, res) => {
+        if (err) return done(err);
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal(exampleGallery.name);
+        expect(res.body.desc).to.equal(exampleGallery.desc);
+        expect(res.body.userID).to.equal(this.tempUser._id.toString());
+        let date = new Date(res.body.created).toString();
+        expect(date).to.not.equal('Invalid Date');
+        done();
+      });
+      });
+
+    //GET test 401 for unauthorized token or no token
+      it('should return status 401 for unauthorized token or no token given', (done) => {
+        request.get(`${url}/api/gallery/${this.tempGallery._id}`)
+        .end((err, res) => {
+          expect(res.status).to.equal(401);
+          done();
+        });
+      });
+    //GET test 404 for bad request with invalid id
+      it('should return 404 for id not found', (done) => {
+        request.get(`${url}/api/gallery/`)
```

Comment:
you should pass a bad ID rather than no ID here in the url. If you make the change to your error handling that I mentioned above (in the GET route), that should work as a 404.